### PR TITLE
[GPU] Fix twin tranformer functional regression.

### PR DESCRIPTION
**Pull request by hyunback**
gemm/FC select_preferred_format select simple format depends on out rank size.
And customized_resnet50_fp16 ACC issue is also solved with this PR.

### Tickets:
 - *105400*

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp [hunk 1]
```diff
@@ -1828,13 +1828,25 @@ void layout_optimizer::select_preferred_formats_for_onednn(program_node& node, d
         for (size_t idx = 0 ; idx < node.get_dependencies().size() ; idx++) {
             if (node.get_dependency(idx).is_constant())
                 continue;
-            node.set_preferred_input_fmt(idx, cldnn::format::bfyx);
+
+            size_t out_rank = node.get_output_layout().get_rank();
+            auto target_format = cldnn::format::bfyx;
+
+            if (out_rank == 5) {
```
[the remark this answers is not in the document]
Comment (hyunback) [reply src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:1835]: oneDNN FC, gemm should use simple format NOT block format now, so for simple, format::get_default_format(rank, is_weights, is_grouped) is enough. I agree to use pre-defined API not if/else newly.

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp [hunk 2]
```diff
@@ -1828,13 +1828,25 @@ void layout_optimizer::select_preferred_formats_for_onednn(program_node& node, d
...
+                target_format = cldnn::format::bfzyx;
+            } else if (out_rank == 6) {
+                target_format = cldnn::format::bfwzyx;
```
[the remark this answers is not in the document]
Comment (hyunback) [reply src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:1838]: Okay I see

File: src/plugins/intel_gpu/tests/test_cases/gemm_gpu_test.cpp [hunk 3]
```diff
@@ -1350,6 +1350,152 @@ class GemmBaseTest : public ::testing::TestWithParam<gemm_params> {
...
+        else
+            return layout{ p.data_type_in2, p.input_format, p.in_shapes.at(2), padding{ pad_ } };
+    }
+
+    cldnn::memory::ptr get_mem(cldnn::layout l) {
+        auto prim = engine.allocate_memory(l);
+        tensor s = l.get_tensor();
+        if (l.data_type == data_types::bin) {
+            VF<int32_t> rnd_vec = generate_random_1d<int32_t>(s.count() / 32, min_random, max_random);
+            set_values(prim, rnd_vec);
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/test_cases/gemm_gpu_test.cpp:1442]: If this is implemented here(not common function), I think we don't need to add such code as it is not running...

File: src/plugins/intel_gpu/tests/test_cases/gemm_gpu_test.cpp [hunk 4]
```diff
@@ -1350,6 +1350,152 @@ class GemmBaseTest : public ::testing::TestWithParam<gemm_params> {
...
+        auto input0_prim = get_mem(get_input_layout(p, 0));
+        auto input1_prim = get_mem(get_input_layout(p, 1));
+
+        network network_ocl(engine, topology_ocl, config_ocl);
+        network network_onednn(engine, topology_onednn, config_onednn);
+
+        network_ocl.set_input_data("input0", input0_prim);
+        network_ocl.set_input_data("input1", input0_prim);
+        network_onednn.set_input_data("input0", input1_prim);
+        network_onednn.set_input_data("input1", input1_prim);
```
[the remark this answers is not in the document]
Comment (hyunback) [reply src/plugins/intel_gpu/tests/test_cases/gemm_gpu_test.cpp:1400]: My mistake, I'll update it

File: src/plugins/intel_gpu/tests/test_cases/gemm_gpu_test.cpp [hunk 5]
```diff
@@ -1350,6 +1350,152 @@ class GemmBaseTest : public ::testing::TestWithParam<gemm_params> {
...
+
+    tolerance = default_tolerance(p.default_type);
+    execute(p);
+}
+#define CASE_GEMM_ONEDNN_FP16_4D { { 2, 3, 2, 2 }, { 2, 3, 2, 2 } }, { 2, 3, 2, 2 }, tensor{ 1 }, tensor{ 0 }, \
+data_types::f16, data_types::f16, data_types::f16, format::bfyx, data_types::f16, format::bfyx
+#define CASE_GEMM_ONEDNN_FP16_5D { { 1, 3, 4, 4, 4 }, { 1, 3, 4, 4, 4 } }, { 1, 3, 4, 4, 4 }, tensor{ 1 }, tensor{ 0 }, \
+data_types::f16, data_types::f16, data_types::f16, format::bfzyx, data_types::f16, format::bfzyx
+#define CASE_GEMM_ONEDNN_FP16_6D { { 2, 3, 5, 5, 3, 2 }, { 2, 3, 5, 5, 3, 2 } }, { 2, 3, 5, 5, 3, 2 }, tensor{ 1 }, tensor{ 0 }, \
+data_types::f16, data_types::f16, data_types::f16, format::bfwzyx, data_types::f16, format::bfwzyx
```
[the remark this answers is not in the document]
Comment (hyunback) [reply src/plugins/intel_gpu/tests/test_cases/gemm_gpu_test.cpp:1492]: tensor(value_type batch_num, value_type feature_num, value_type x, value_type y, value_type z, value_type w)  in tensor.hpp

File: src/plugins/intel_gpu/tests/test_cases/gemm_gpu_test.cpp (hunk 3 above)
Comment (hyunback) [reply src/plugins/intel_gpu/tests/test_cases/gemm_gpu_test.cpp:1442]: I'll remove it

File: src/plugins/intel_gpu/tests/test_cases/gemm_gpu_test.cpp [hunk 6]
```diff
@@ -1350,6 +1350,152 @@ class GemmBaseTest : public ::testing::TestWithParam<gemm_params> {
...
+        } else if (l.data_type == data_types::i8 || l.data_type == data_types::u8) {
+            VF<uint8_t> rnd_vec = generate_random_1d<uint8_t>(s.count(), min_random, max_random);
+            set_values(prim, rnd_vec);
+        } else if (l.data_type == data_types::f16) {
+            VF<uint16_t> rnd_vec = generate_random_1d<uint16_t>(s.count(), -1, 1);
+            set_values(prim, rnd_vec);
+        } else {
+            VF<float> rnd_vec = generate_random_1d<float>(s.count(), -1, 1);
+            set_values(prim, rnd_vec);
+        }
```
[the remark this answers is not in the document]
Comment (hyunback) [reply src/plugins/intel_gpu/tests/test_cases/gemm_gpu_test.cpp]: applied it

**Review by isanghao — APPROVED** [review]

**Pull request merged**